src/network/networkd-link.c

```diff
@@ -99,6 +99,21 @@ static bool link_ipv6ll_enabled(Link *link) {
         return link->network->link_local & ADDRESS_FAMILY_IPV6;
 }
 
+static bool link_ipv6_enabled(Link *link) {
+        assert(link);
+
+        if (!socket_ipv6_is_supported())
+                return false;
+
+        if (link_dhcp6_enabled(link) && link_ipv6ll_enabled(link))
+                return true;
```

Comment:
https://en.wikipedia.org/wiki/De_Morgan's_laws :)

@poettering's comment translates to:
```
 if (!socket_ipv6_is_supported())
                return false;
return link_dhcp6_enabled(link) || 
          link_ipv6ll_enabled(link) ||
          network_has_static_ipv6_addresses(link->network);
```